src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -284,206 +425,247 @@ private void hoeIfAble(final BlockPos position, final Field field)
-        {
-            requestSeeds = false;
-        }
-
-        return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.up()).getBlock() instanceof BlockCrops)
-                 && !(world.getBlockState(position).getBlock() instanceof BlockHutField) && world.getBlockState(position).getBlock() == Blocks.FARMLAND;
-    }
-
-    /**
-     * Plants the crop at a given location.
-     *
-     * @param item     the crop.
-     * @param position the location.
-     */
-    private boolean plantCrop(Item item, @NotNull BlockPos position)
+    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final Field field)
     {
-        final int slot = worker.findFirstSlotInInventoryWith(item);
-        if (slot == -1)
-        {
-            return false;
-        }
-        else
-        {
-            @NotNull final IPlantable seed = (IPlantable) item;
-            world.setBlockState(position.up(), seed.getPlant(world, position));
-            getInventory().decrStackSize(slot, 1);
-            requestSeeds = false;
-            //Flag 1+2 is needed for updates
-            return true;
-        }
+        return !field.isNoPartOfField(world, position)
+                 && !BlockUtils.isBlockSeed(world, position.up())
+                 && !(world.getBlockState(position).getBlock() instanceof BlockHutField)
+                 && (world.getBlockState(position).getBlock() == Blocks.DIRT || world.getBlockState(position).getBlock() == Blocks.GRASS);
     }
 
     /**
-     * Resets the basic variables of the class.
+     * Sets the hoe as held item.
      */
-    private void resetVariables()
+    private void equipHoe()
     {
-        requestSeeds = true;
-        shouldDumpInventory = true;
-        shouldTryToGetSeed = true;
+        worker.setHeldItem(getHoeSlot());
     }
+    
+    private int dist = 0;
+    private boolean horizontal = false;
+    private int totalDis = 1;
 
     /**
-     * Terminates the planting process and resets the task.
+     * Handles the offset of the field for the farmer.
      *
-     * @param buildingFarmer the building of the farmer.
-     * @param field          the field being planted.
-     * @return the next state.
+     * @param field the field object.
+     * @return true if successful.
      */
-    @NotNull
-    private AIState terminatePlanting(@NotNull final BuildingFarmer buildingFarmer, @NotNull final Field field)
+    private boolean handleOffset(@NotNull final Field field)
     {
-        if (requestSeeds)
+        if (workingOffset == null)
         {
-            chatSpamFilter.talkWithoutSpam("entity.farmer.NeedSeed", field.getSeed().getItemStackDisplayName(new ItemStack(field.getSeed())));
+            workingOffset = new BlockPos(1, 0,0);
+            totalDis = 1;
+            dist = 0;
         }
         else
         {
-            buildingFarmer.getCurrentField().setNeedsWork(false);
+            final int absZ = Math.abs(workingOffset.getZ());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "absZ" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "absZ". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)